include/boost/beast/core/impl/dynamic_buffer.hpp

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright (c) 2020 Richard Hodges (hodges.r@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_CORE_IMPL_DYNAMIC_BUFFER_HPP
+#define BOOST_BEAST_CORE_IMPL_DYNAMIC_BUFFER_HPP
+
+#include <boost/beast/core/dynamic_buffer.hpp>
+#include <memory>
+
+namespace boost {
+namespace beast {
+
+template<class DynamicBuffer_v0>
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+dynamic_buffer_v0_proxy(
+    DynamicBuffer_v0& storage) noexcept
+: storage_(std::addressof(storage))
+{
+    boost::ignore_unused(storage_->prepare(0));
```

Comment:
I am not a fan of this use of `ignore_unused`  to discard the return value.

Reply:
This should be rolled up into a member function `normalize()` (or `normalise()`)